src/OpenLoco/src/Ui/WindowManager.cpp

```diff
@@ -1466,6 +1467,103 @@ namespace OpenLoco::Ui::WindowManager
         invalidateWidget(window->type, window->number, widgetIndex);
     }
 
+    // Allow mouse wheel scrolling to manipulate stepper widgets and tool sizes
+    static bool stepperWheelInput(Window& w, WidgetIndex_t widgetIndex, int32_t wheel)
+    {
+        auto widgetType = w.widgets[widgetIndex].type;
+
+        // If we aren't scrolling on a tool preview or stepper textbox, lower the widget index
+        // once or twice to try to find the parent widget. This allows scrolling on the
+        // increase/decrease buttons themselves, too.
+        int32_t attempts = 0;
+        while (widgetType != WidgetType::wt_3 && widgetType != WidgetType::textbox && widgetIndex > 0)
+        {
+            switch (widgetType)
+            {
+                case WidgetType::toolbarTab: // + and - for tool widget
+                case WidgetType::button:     // + and - for stepper widget
+                {
+                    if (attempts > 0)
+                    {
+                        // Verify that the previous button was of the same type
+                        auto previousType = w.widgets[widgetIndex + 1].type;
+                        if (previousType != widgetType)
+                        {
+                            return false;
+                        }
+                    }
+                    break;
+                }
+                default:
+                    // The widget type is not an increment or decrement button
+                    return false;
+            }
+
+            attempts++;
+            if (attempts > 2)
+            {
+                // We've looked two widgets up, and no tool preview or stepper widget was found
+                return false;
+            }
+
+            widgetIndex--;
```

Comment:
Is this safe to do? If the widgetIndex is 0 this will roll over.

Reply:
That's a good point. In practice, it won't happen as the first three widgets are always frame/panel/close, but we should add an assertion at least.

Reply:
Actually there's already a `widgetIndex > 0` in the while loop condition, so this won't be a problem.